nan.h

```diff
@@ -1527,26 +1755,28 @@ class NanObjectWrap : public node::ObjectWrap {
 template <typename T, typename P>
 class NanWeakCallbackData {
  public:  // constructors
+  inline ~NanWeakCallbackData();
+
+ private:  // constructors
+  NAN_DISALLOW_ASSIGN_COPY_MOVE(NanWeakCallbackData)
+
+ public:
   typedef void (*Callback)(
       NanWeakCallbackData & data  // NOLINT(runtime/references)
       );
-  NanWeakCallbackData(v8::Handle<T> handle, P* param, Callback cb)
-    : parameter(param), callback(cb) {
-    NanAssignPersistent(persistent, handle);
-    Revive();
+  NanWeakCallbackData(v8::Persistent<T> *handle, P* param, Callback cb)
+    : persistent(), parameter(param), callback(cb) {
+    std::memcpy(&persistent, handle, sizeof(persistent));
```

Comment:
To clarify, the situation is this:
```c++
{
  Isolate * isolate; /*...*/
  Local<Value> val; /*...*/
  Persistent<Value> p(isolate, val);
  int * i = new int(0);
  p.SetWeak(isolate, i, weak);
} // p goes out of scope and is destroyed, but its storage cell remains

void weak(...) {
  // dispose p
}
```
This would normally require having p globally accessible (directly or indirectly), to be able to reference it from the weak callback. I embed a copy of p in the data pointer supplied to the weak callback, so I may dispose it through that. Note that I do not want a copy made of the storage cell p refers to, just of the handle.